components/raftstore/src/store/config.rs

```diff
@@ -334,6 +334,13 @@ pub struct Config {
     #[online_config(hidden)]
     // Interval to check if need to request snapshot.
     pub check_request_snapshot_interval: ReadableDuration,
+
+    /// Make raftstore v1 learners compatible with raftstore v2 by:
+    /// * Recving tablet snapshot from v2.
+    /// * Responsing GcPeerRequest from v2.
+    #[doc(hidden)]
+    #[online_config(hidden)]
+    pub enable_v2_compatible_learner: bool,
```

Comment:
Internally in the code, we have v1 or v2 concept. However, in the configuration file, we don't expose this to end-user. 
End user will see raft-kv or partitioned-raft-kv as engine type, not raft-kv2. 
So we should add an alias to this config,  like enable-partitioned-raft-kv-compatible-learner